3drepo/3drepo.io
# Issue 5188 - Ticket filters: filters bar [base branch]

This fixes #5188 

#### Description
The scope of this branch was originally to display the filter chips inside a filter bar component. As such, focus primarily on the chips:
- the grouping by module (if module is an empty string, that's the top panel module and no header should be displayed for that section)
- the text-overflow (the operator should always be visible, the property name should be the first to break, then the value might break)
- when there is more than one value for a given pill, it should state "x values",
- etc.

On top of that, to make life easier, I provided a way to create new filters. You may notice 3 inputs: filter, value, isDate.
- isDate should only be clicked if the provided value could be passed to the `Date` constructor (eg. `10/12/2024` or `1728687600000`)
- value, which will always be treated as a string for now, but that is good enough for testing purposes
- filter: this should be 3 elements (module, property, operation) joined together with a dot
> e.g
> - "module1.prop1.ex" Would be read as "module1, property prop1, operator exist
> - ".assignees.ss" Would be read as "top panel (see before the first `.` there is nothing, so empty module), property assigneed, operator contains)

The list of valid operators can be found here https://github.com/3drepo/3drepo.io/issues/5244 or in the code, in the file `cardFilters.types.ts`

#### Test cases
- Make sure you can collapse/expand the filters section, and that the toggle expansion button is only available when the filters bar is tall enough. The component should be scrollable if necessary (even when collapsed)
> Testing the sizing, keep in mind what the final outlook should be, that is, the top "form" to create a new filter should not be there
- Test filters deletion (single or clear all) as well as text-overflowing scenarios
- Check grouping by module


#### Bugs list (Updated: 25/03/2025)
- [x] [Image preview is broken](https://github.com/3drepo/3drepo.io/pull/5252#issuecomment-2720555535) (Daniel)
- [x] [Toggle button disappears when adding/removing filter (collapsed filters bar)](https://github.com/3drepo/3drepo.io/pull/5252#issuecomment-2751531555) (Daniel)
- [x] [Tickets card is opened when entering federation/container](https://github.com/3drepo/3drepo.io/pull/5252#issuecomment-2751545254) (Daniel)
- [x] [Scrollbar STILL not working properly when updating filter. Cannot reach the "Update" button](https://github.com/3drepo/3drepo.io/pull/5252#issuecomment-2751652442) (Daniel)
- [x] [Inconsistent behaviour on changing criteria when updating filters.](https://github.com/3drepo/3drepo.io/pull/5252#issuecomment-2751956821) (Daniel)
